### DISCUSSION TITLE
More options for subscription gifting

### DISCUSSION THREAD

**Discussion by Logan9312** [gameserverapp/Platform#721] [Suggestions]
This is kind of a combo suggestion, there's two approaches I think would be good to fix the issue I'm having.

# The Problem
I sell subscriptions on my server that come with a whitelisted slot for a premium map. This works great for single person subs, but I also have tiers where you can buy slots for other members (For example you can buy a tier that gives access to your whole tribe). Currently I'm going to setup a discord notification for other tiers and users can tell us manually who they want whitelisted

## Solution 1: Gifting
Allow users to purchase a subscription on behalf of another user, so instead of having a 1 slot tier and a 2 slot tier for example, someone could subscribe to a one slot tier for themselves, and gift a 1 slot tier to someone else along with all the benefits. 

## Solution 2: Select User Automation
When you configure something on GSA, you can choose commands that run in reference to the current user purchasing, for example whitelist the current user. I propose a new setting to allow the purchaser to select a new user from the cluster to have a specific benefit apply to. For example in the two slot combo, it would auto whitelist the purchaser, but also allow the purchaser to select another user in the cluster to have whitelisted.

For example, I have added a new option to the whitelist option showcasing what it could look like. The options being "Purchaser" or "Allow purchaser to select"
<img width="760" alt="image" src="https://github.com/gameserverapp/Platform/assets/87048158/c5b9ed28-ba20-4d89-8275-e32f2ea5db15">

**Comment by mvaessen** [comment]
i think there already is a suggestion for being able to gift stuff from the shop / supporter tier. But gifting half of stuff seems a bit overcomplicating.
